tests/components/withings/test_common.py

```diff
@@ -15,6 +15,13 @@
 from homeassistant.exceptions import PlatformNotReady
 from homeassistant.util import dt
 
+DEFAULT_TIME_ZONE = dt.DEFAULT_TIME_ZONE
+
+
+def teardown():
```

Comment:
I would recommend we instead patch `"homeassistant.util.dt.DEFAULT_TIME_ZONE"` or whatever path is needed in the code under test, and not call the core config function at all in the test.

Patches are undone by default when they end.

Reply:
You have a gift for reviewing merged PR's sir! 😂

Reply:
I don't see how we can patch this.

Patching what you suggest will run into this:

```
    def now(time_zone: Optional[dt.tzinfo] = None) -> dt.datetime:
        """Get now in specified time zone."""
>       return dt.datetime.now(time_zone or DEFAULT_TIME_ZONE)
E       TypeError: tzinfo argument must be None or of a tzinfo subclass, not type 'MagicMock'
```

Which makes sense.


We can avoid calling the core config function using the following patch:

```patch
diff --git a/tests/components/withings/test_common.py b/tests/components/withings/test_common.py
index 6c8c0a4c3..2dd024908 100644
--- a/tests/components/withings/test_common.py
+++ b/tests/components/withings/test_common.py
@@ -10,7 +10,6 @@ from homeassistant.components.withings.common import (
     NotAuthenticatedError,
     WithingsDataManager,
 )
-from homeassistant.config import async_process_ha_core_config
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import PlatformNotReady
 from homeassistant.util import dt
@@ -121,10 +120,7 @@ async def test_data_manager_update_sleep_date_range(
     hass: HomeAssistant, data_manager: WithingsDataManager,
 ) -> None:
     """Test method."""
-    await async_process_ha_core_config(
-        hass=hass, config={"time_zone": "America/Los_Angeles"}
-    )
-
+    dt.set_default_time_zone(dt.get_time_zone("America/Los_Angeles"))
     update_start_time = dt.now()
     await data_manager.update_sleep()
```

Still need the teardown, but at least it will remove the `async_process_ha_core_config` call.

Reply:
We need to set the `new` argument in the patch to the correct type.

Reply:
doh! 🤦‍♂ 

https://github.com/home-assistant/home-assistant/pull/30326